Helper/Invoice.php

```diff
@@ -249,14 +262,32 @@ public function handleCaptureWebhook(Order $order, Notification $notification):
     {
         $invoiceFactory = $this->adyenInvoiceFactory->create();
         $adyenInvoice = $this->adyenInvoiceResourceModel->getAdyenInvoiceByCaptureWebhook($order, $notification);
+        $isFullAmountCaptured = $this->adyenDataHelper->originalAmount($notification->getAmountValue(), $notification->getAmountCurrency()) >= $order->getBaseGrandTotal();
 
-        if (is_null($adyenInvoice)) {
-            throw new \Exception(sprintf(
-                'Unable to find adyen_invoice linked to original reference %s, psp reference %s, and order %s',
+        if (is_null($adyenInvoice) && $order->canInvoice()) {
+                if($isFullAmountCaptured) {
+                    $this->createInvoiceFromWebhook($order, $notification);
```

Comment:
I think we have to update `$adyenInvoice`, otherwise it will remain `null` and fail later (e.g. line 294):
```suggestion
                    $adyenInvoice = $this->createInvoiceFromWebhook($order, $notification);
```